arch/arm/src/s32k3xx/s32k3xx_edma.c

```diff
@@ -0,0 +1,1626 @@
+/****************************************************************************
+ * arch/arm/src/s32k3xx/s32k3xx_edma.c
+ *
+ *   Copyright (C) 2019, 2021 Gregory Nutt. All rights reserved.
+ *   Copyright 2022 NXP
+ *   Authors: Gregory Nutt <gnutt@nuttx.org>
+ *            David Sidrane <david.sidrane@nscdg.com>
+ *            Peter van der Perk <peter.vanderperk@nxp.com>
+ *
+ * This file was leveraged from the NuttX S32K1 port.  Portions of that eDMA
+ * logic derived from NXP sample code which has a compatible BSD 3-clause
+ * license:
+ *
+ *   Copyright (c) 2015, Freescale Semiconductor, Inc.
+ *   Copyright 2016-2017 NXP
```

Comment:
can we use Apache licence?

Reply:
It's based on the original code here
https://github.com/apache/incubator-nuttx/blob/master/arch/arm/src/s32k1xx/s32k1xx_edma.c
With some backports from imxrt which @davids5 wrote.
I rather don't like to change the license of a file, but If @davids5 & @gregory-nutt can confirm they're okay with it I can change it.

Reply:
Fine by me - We have all already lost the credits for all the work we have done.